core-bundle/tests/EventListener/GlobalsMapListenerTest.php

```diff
@@ -18,30 +18,44 @@
 class GlobalsMapListenerTest extends TestCase
 {
     /**
-     * @param array|string|null $existing
-     * @param array|string      $new
-     *
      * @dataProvider getValuesData
-     *
-     * @runInSeparateProcess
-     * @preserveGlobalState disabled
      */
-    public function testMergesTheValuesIntoTheGlobalsArray(string $key, $existing, $new): void
+    public function testMergesTheValuesIntoTheGlobalsArray(array $existing, array $new, array $expected): void
     {
-        $GLOBALS[$key] = $existing;
+        $GLOBALS['TL_CTE'] = $existing;
 
-        $listener = new GlobalsMapListener([$key => $new]);
+        $listener = new GlobalsMapListener(['TL_CTE' => $new]);
         $listener->onInitializeSystem();
 
-        $this->assertSame($new, $GLOBALS[$key]);
+        $this->assertSame($expected, $GLOBALS['TL_CTE']);
+
+        unset($GLOBALS['TL_CTE']);
     }
 
     public function getValuesData(): \Generator
     {
-        yield ['foo', null, 'bar'];
-        yield ['foo', 'bar', 'baz'];
-
-        yield ['TL_CTE', null, ['foo' => 'bar']];
-        yield ['TL_CTE', ['foo' => 'bar'], ['foo' => 'baz']];
+        yield 'add single' => [
+            [],
+            ['text' => 'HeadlineFragment'],
+            ['text' => 'HeadlineFragment'],
+        ];
+
```

Comment:
```php
        yield 'overwrite single' => [
            ['text' => 'Headline'],
            ['text' => 'HeadlineFragment'],
            ['text' => 'HeadlineFragment'],
        ];
```

This would fail now because of the change in `array_replace_recursive()`?

Reply:
IMHO this is intended to fail, but maybe I'm misunderstanding the test case? The idea is: If there is already a legacy element registered, adding a fragment won't overrule it. This way you can add a `$GLOBALS['TL_CTE']['texts]['text']` entry to disable the new `TextController` fragment.